contrib/pyln-proto/pyln/proto/wire.py

```diff
@@ -59,18 +58,19 @@ def decryptWithAD(k, n, ad, ciphertext):
 
 class PrivateKey(object):
     def __init__(self, rawkey):
-        assert len(rawkey) == 32 and isinstance(rawkey, bytes)
+        if not isinstance(rawkey, bytes):
+            raise TypeError(f"rawkey must be bytes, {type(rawkey)} received")
+        elif len(rawkey) != 32:
+            raise ValueError(f"rawkey must be 32-byte long. {len(rawkey)} received")
+
         self.rawkey = rawkey
-        rawkey = int(hexlify(rawkey), base=16)
-        self.key = ec.derive_private_key(rawkey, ec.SECP256K1(),
-                                         default_backend())
+        self.key = coincurve.PrivateKey.from_int(int(hexlify(rawkey), base=16))
 
     def serializeCompressed(self):
-        return self.key.private_bytes(serialization.Encoding.Raw,
-                                      serialization.PrivateFormat.Raw, None)
+        return hexlify(self.key.to_hex())
```

Comment:
self.key.format()?

Reply:
There's no format for PrivateKey, only for PublicKey in coincurve. 

I can do ``bytes.fromhex(self.key.to_hex())``, but that'll be equivalent to ``self.rawkey`` and ``self.key.secret``.

Is this the expected return though, the current version of the method doesn't work AFAICT so not sure if the raw key is expected or a base58 encoding. If so, any of the last two should do.

Reply:
I've gone for `self.key.secret` in 0cbe96287c932134eb7f94a97e92765536e1821c, but can rebase and change it if the expected return format is different

Reply:
you should fold in your later changes, such that this patch does not introduce any new uses of hexlify.